### PR TITLE
Workspace types

### PR DESCRIPTION
**Pull request by coderbydesign**
## Link(s) to Jira
- https://issues.redhat.com/browse/RHCLOUD-35289

## Description of Intent of Change(s)
This adds a `type` column on the `Workspace` model in order to identify in the
code and API, which type of workspace an object is:

- `root`: top-level workspace for a tenant
- `default`: the default workspace for a tenant
- `standard`: any custom non-root/default workspace

Constraints are added to ensure a single root and single default workspace per tenant.

The `type` field is also return on workspace responses (but disallowed from create and update operations via the API):
```json
{
  "uuid": "3fa85f64-5717-4562-b3fc-2c963f66afa6",
  "parent_id": "3fa85f64-5717-4562-b3fc-2c963f66afa6",
  "type": "root",
  "name": "Workspace A",
  "description": "Description of Workspace A",
  "created": "2024-09-27T18:22:39.738Z",
  "modified": "2024-09-27T18:22:39.738Z"
}
```

## Local Testing
How can the feature be exercised?
How can the bug be exploited and fix confirmed?
Is any special local setup required?

## Checklist
- [x] if API spec changes are required, is the spec updated?
- [ ] are there any pre/post merge actions required? if so, document here.
- [x] are theses changes covered by unit tests?
- [ ] if warranted, are documentation changes accounted for?
- [ ] does this require migration changes?
  - [ ] if yes, are they backwards compatible?
- [ ] is there known, direct impact to dependent teams/components?
  - [ ] if yes, how will this be handled?

## Secure Coding Practices Checklist Link
- https://github.com/RedHatInsights/secure-coding-checklist

## Secure Coding Practices Checklist
- [ ] Input Validation
- [ ] Output Encoding
- [ ] Authentication and Password Management
- [ ] Session Management
- [ ] Access Control
- [ ] Cryptographic Practices
- [ ] Error Handling and Logging
- [ ] Data Protection
- [ ] Communication Security
- [ ] System Configuration
- [ ] Database Security
- [ ] File Management
- [ ] Memory Management
- [ ] General Coding Practices

### CODE REVIEW

File: rbac/management/workspace/model.py [hunk 1]
```diff
@@ -27,17 +28,35 @@
...
+            UniqueConstraint(
+                fields=["tenant_id", "type"],
+                name="unique_default_root_workspace_per_tenant",
+                condition=Q(type__in=["root", "default"]),
+            )
+        ]
+
+    def save(self, *args, **kwargs):
+        """Override save on model to enforce validations."""
+        self.full_clean()
```
Comment (astrozzc) [inline rbac/management/workspace/model.py:58]: Curious what happens if we don't add this one? Will the save pass the constrain validation?

**Review by astrozzc — APPROVED** [review]
LGTM

**Review by alechenninger — APPROVED** [review]
LGTM!

File: rbac/management/workspace/model.py (hunk 1 above)
Comment (coderbydesign) [reply rbac/management/workspace/model.py:58]: The uniqueness constraints would still trigger, but the type choices would not be enforced, so you could create a workspace of type "foo".

**Review by lpichler — APPROVED** [review]

**Pull request merged**